src/main/java/com/notably/logic/correction/AbsolutePathCorrectionEngine.java

```diff
@@ -0,0 +1,148 @@
+package com.notably.logic.correction;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Queue;
+import java.util.stream.Collectors;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.commons.core.path.exceptions.InvalidPathException;
+import com.notably.logic.correction.distance.EditDistanceCalculator;
+import com.notably.logic.correction.distance.LevenshteinDistanceCalculator;
+import com.notably.model.Model;
+import com.notably.model.block.BlockTree;
+import com.notably.model.block.BlockTreeItem;
+
+/**
+ * Represents a correction engine that works on {@link AbsolutePath}s
+ */
+public class AbsolutePathCorrectionEngine implements CorrectionEngine<AbsolutePath> {
+    private final EditDistanceCalculator editDistanceCalculator;
+    private final Model model;
+    private final int distanceThreshold;
+
+    public AbsolutePathCorrectionEngine(Model model, int distanceThreshold) {
+        this(new LevenshteinDistanceCalculator(), model, distanceThreshold);
+    }
+
+    public AbsolutePathCorrectionEngine(EditDistanceCalculator editDistanceCalculator,
+            Model model, int distanceThreshold) {
+        Objects.requireNonNull(editDistanceCalculator);
```

Comment:
Can I ask why you omit the nullity check for `distanceThreshold`?

Reply:
`distanceThreshold` is an integer, cannot be null!